src/core/net/srp_server.cpp

```diff
@@ -633,7 +639,7 @@ void Server::Start(void)
     return;
 }
 
-void Server::PrepareSocket(void)
+Error Server::PrepareSocket(void)
```

Comment:
How about the following approach? 

- We can read and set the `mPort` from `Settings` from constructor.
- `SelectPort()` will be simpler and will move to the next port value (without preparing socket or reading it from `Settings`).
- We call `PrepareSocket()` from `Start()` as before.
- But if `PrepareSocket()` fails, we call `Disable()` immediately followed by `Enable()`. 
- This way we ensure last published entry is removed from netdata before a new port is selected and we wait for it to be published before calling `Start()`.
- This approach should preserve the original operational flow (when socket is prepared) similar to before.

Reply:
Sounds reasonable to me. Thanks for the suggestion! Updated.

Reply:
Good idea. I was thinking we may need a method to call from `Instance::AfterInit()` (like in case of `MleRouter` restoring)  but the model you added to track if port is uninitialized is better and simpler. 👍